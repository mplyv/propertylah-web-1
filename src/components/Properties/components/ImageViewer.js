import styles from '../property.module.css'
import { useState } from 'react'

const ImageViewer = (props)=>{
    const [image, setImage] = useState(0)
    return(
        <>
            <img data-testid={'scrollingImage'} className={styles.image} src={props.source[image]} alt='image1' />   
            {image === 0? <button data-testid={'prevButton'} className={styles.imageButtonPrev} style={{padding:'0px'}} onClick={(e)=>{e.stopPropagation()}}><>&lt;</></button>
            :<button data-testid={'prevButton'} className={styles.imageButtonPrev} style={{padding:'0px'}} onClick={(e)=>{e.stopPropagation(); setImage(image-1)}}><>&lt;</></button>}
            {image === props.source.length-1? <button data-testid={'nextButton'} className={styles.imageButtonNext} style={{padding:'0px', left:'538px'}} onClick={(e)=>{e.stopPropagation()}}><>&gt;</></button>
            :<button data-testid={'nextButton'} className={styles.imageButtonNext} style={{padding:'0px', left:'538px'}} onClick={(e)=>{e.stopPropagation(); setImage(image+1)}}><>&gt;</></button>}
        </>
    )
}


export default ImageViewer