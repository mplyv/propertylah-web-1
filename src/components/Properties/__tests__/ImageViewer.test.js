import {render, screen, cleanup} from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import ImageViewer from '../components/ImageViewer'
import renderer from 'react-test-renderer'

//* cleanup after each test
afterEach(() => {
    cleanup();
});

let source=[
    '/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-1.jpg',
    '/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-2.jpg',
    '/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-3.jpg',
    '/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-4.jpg'
]

let onClick = jest.fn();

test('First image should render',()=>{
    render(<ImageViewer source={source} />);
    const image = screen.getByTestId(`scrollingImage`);
    expect(image).toBeInTheDocument();
    expect(image.src).toBe('http://localhost/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-1.jpg');
})

test('buttons should render',()=>{
    render(<ImageViewer source={source} />);
    const prevButton = screen.getByTestId(`prevButton`);
    const nextButton = screen.getByTestId(`nextButton`);
    expect(prevButton).toBeInTheDocument();
    expect(nextButton).toBeInTheDocument();
})

test('On initial load, clicking next button should go to next image', async()=>{
    render(<ImageViewer source={source}/>);
    const user = userEvent.setup()
    const image = screen.getByTestId(`scrollingImage`);
    const nextButton = screen.getByTestId(`nextButton`);
    await user.click(nextButton);
    expect(image.src).toBe('http://localhost/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-2.jpg');
})

test('On initial load, clicking prev button should not change image', async()=>{
    render(<ImageViewer source={source}/>);
    const user = userEvent.setup()
    const image = screen.getByTestId(`scrollingImage`);
    const prevButton = screen.getByTestId(`prevButton`);
    await user.click(prevButton);
    expect(image.src).toBe('http://localhost/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-1.jpg');
})

test('If only 1 image, both buttons do nothing', async()=>{
    render(<ImageViewer source={['/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-4.jpg']}/>);
    const user = userEvent.setup()
    const image = screen.getByTestId(`scrollingImage`);
    const prevButton = screen.getByTestId(`prevButton`);
    const nextButton = screen.getByTestId(`nextButton`);
    await user.click(nextButton);
    expect(image.src).toBe('http://localhost/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-4.jpg');
    await user.click(prevButton);
    expect(image.src).toBe('http://localhost/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-4.jpg');
})

test('match source snapshot',()=>{
    const tree = renderer.create(<ImageViewer source={source}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
