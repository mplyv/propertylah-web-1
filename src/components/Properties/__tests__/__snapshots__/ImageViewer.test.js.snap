// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match source snapshot 1`] = `
Array [
  <img
    alt="image1"
    className="image"
    data-testid="scrollingImage"
    src="/assets/property-images/Sky-Vue-Ang-Mo-Kio-Bishan-Thomson-Singapore-1.jpg"
  />,
  <button
    className="imageButtonPrev"
    data-testid="prevButton"
    onClick={[Function]}
    style={
      Object {
        "padding": "0px",
      }
    }
  >
    &lt;
  </button>,
  <button
    className="imageButtonNext"
    data-testid="nextButton"
    onClick={[Function]}
    style={
      Object {
        "left": "538px",
        "padding": "0px",
      }
    }
  >
    &gt;
  </button>,
]
`;
